src/test/ui/issues/issue-59488.rs

```diff
@@ -31,4 +31,5 @@ fn main() {
     //~^ ERROR binary operation `==` cannot be applied to type `fn(usize) -> Foo {Foo::Bar}` [E0369]
     //~| ERROR `fn(usize) -> Foo {Foo::Bar}` doesn't implement `Debug` [E0277]
     //~| ERROR `fn(usize) -> Foo {Foo::Bar}` doesn't implement `Debug` [E0277]
+    //~| ERROR `fn(usize) -> Foo {Foo::Bar}` doesn't implement `Debug` [E0277]
```

Comment:
This now duplciates an error. Can that be avoided?

Reply:
I don't think so. The expression is still used to create the same panic message as before, but now also used to form a `&dyn Debug`.

I'd be happy to avoid this if possible though.

Reply:
hmm, maybe it can be done by using the same `&dyn Debug` also for the panic message. Not sure if that is desirable though.

Reply:
I modified the macro to assign `left_val` and `right_val` to a `&dyn Debug` once, and then use that for the rest of the code. The extra error is gone that way.

Interestingly, the compiler now only gives an error once per type. So if left and right are different types (and both don't implement `Debug`), it gives an error for both. If they are the same type, only one error is generated. This seems like nice behaviour, so I added a UI test for it.